c_glib/arrow-glib/file-system.cpp

```diff
@@ -0,0 +1,1050 @@
+}
+
+/**
+ * garrow_file_system_get_type_name:
+ * @file_system: A #GArrowFileSystem.
+ *
+ * Returns: The name of file system type.
+ *
+ * Since: 1.0.0
+ */
+const gchar *
+garrow_file_system_get_type_name(GArrowFileSystem *file_system)
+{
+  auto arrow_file_system = garrow_file_system_get_raw(file_system);
+  return arrow_file_system->type_name().c_str();
+}
+
+/**
+ * garrow_file_system_get_target_stats:
+ * @file_system: A #GArrowFileSystem.
+ * @path: The path of the target.
+ * @error: (nullable): Return location for a #GError or %NULL.
+ *
+ * Get statistics for the given target.
+ *
+ * Any symlink is automatically dereferenced, recursively.
+ * A non-existing or unreachable file returns an OK status and has
+ * a #GArrowFileType of value %GARROW_FILE_TYPE_NON_EXISTENT.
+ * An error status indicates a truly exceptional condition
+ * (low-level I/O error, etc.).
+ *
+ * Returns: (nullable) (transfer full): A #GArrowFileStats
+ *
+ * Since: 1.0.0
+ */
+GArrowFileStats *
+garrow_file_system_get_target_stats(GArrowFileSystem *file_system,
+                                    const gchar *path,
+                                    GError **error)
+{
+  auto arrow_file_system = garrow_file_system_get_raw(file_system);
+  auto arrow_result = arrow_file_system->GetTargetStats(path);
+  if (garrow::check(error, arrow_result, "[filesystem][get_target_stats]")) {
+    auto arrow_file_stats = std::make_shared<arrow::fs::FileStats>(arrow_result.ValueOrDie());
+    return garrow_file_stats_new_raw(&arrow_file_stats);
+  } else {
+    return NULL;
+  }
+}
+
+namespace {
+
+inline GList *
+garrow_file_stats_list_from_result(arrow::Result<std::vector<arrow::fs::FileStats>>&& arrow_result,
+                                   GError **error,
+                                   const gchar *context)
+{
+  if (garrow::check(error, arrow_result, context)) {
+    auto arrow_file_stats_vector = arrow_result.ValueOrDie();
+    GList *file_stats_list = NULL;
+    for (auto arrow_file_stats : arrow_file_stats_vector) {
+      auto arrow_file_stats_ptr = std::make_shared<decltype(arrow_file_stats)>(arrow_file_stats);
+      auto file_stats = garrow_file_stats_new_raw(&arrow_file_stats_ptr);
+      file_stats_list = g_list_prepend(file_stats_list, file_stats);
+    }
+    return g_list_reverse(file_stats_list);
+  } else {
+    return NULL;
+  }
+}
+
+}
+
+/**
+ * garrow_file_system_get_target_stats_list:
+ * @file_system: A #GArrowFileSystem.
+ * @paths: The paths of the targets.
+ * @n_paths: The number of items in @paths.
+ * @error: (nullable): Return location for a #GError or %NULL.
+ *
+ * Get statistics same as garrow_file_system_get_target_stats() for the
+ * given many targets at once.
+ *
+ * Returns: (element-type GArrowFileStats) (transfer full):
+ *   A list of #GArrowFileStats
+ *
+ * Since: 1.0.0
+ */
+GList *
+garrow_file_system_get_target_stats_list(GArrowFileSystem *file_system,
```

Comment:
We use suffix for function name that has some variants like `garrow_decimal128_new_string()` and `garrow_decimal128_new_integer()`.
How about `garrow_file_system_get_target_stats_paths()`?